pandas/core/groupby/groupby.py

```diff
@@ -1781,6 +1818,23 @@ def mean(self, numeric_only: bool | lib.NoDefault = lib.no_default):
             Include only float, int, boolean columns. If None, will attempt to use
             everything, then use only numeric data.
 
+        engine : str, default None
```

Comment:
im not wild about adding to the API.  is there a viable way to just detect if numba is available and if so always use it?

Reply:
`maybe_use_numba` can detect if the user has specified the numba option as a `config` option. 

Though, these keyword options are consistent with the reduction methods on windows ops and my plan to also expose this regular `DataFrame/Series` reduction methods as well.

Reply:
OT: should make this into a template that can append rather than copy-pasting